core/cli/src/informant.rs

```diff
@@ -85,6 +85,7 @@ pub fn start<C>(service: &Service<C>, exit: ::exit_future::Exit, handle: TaskExe
 			telemetry!(
 				SUBSTRATE_INFO;
 				"system.interval";
+				"network_state" => format!("{:?}", network.network_state()),
```

Comment:
Wouldn't a reference work here? From what I can see `NetworkState` implements `Serialize`, so it should be printable to JSON. I can probably work with a debug print, but I don't think it makes for the best API.

Reply:
Something like `serde_json::to_string(&network.network_state())`.

Reply:
I think @maciejhirsz wanted to propose:
```suggestion
				"network_state" => &network.network_state(),
```

Reply:
If you pass the `network_state` directly, whether by value or reference, it complains in that way:
`the trait `slog::Value` is not implemented for `substrate_network_libp2p::NetworkState``.

From the docs I read "Without an implementation of Value for your type you must emit using either the ? "debug", % "display" or SerdeValue (if you have the nested-values feature enabled) formatters.", which made me try using `format!`, and it works. 

I would say it's probably the right thing to do for now, since we don't know yet what we're going to do with the data. An implementation of `slog::Value` for `NetworkState` could follow once we know more about what we want...

Would using `serde_json::to_string` offer any advantage over `format!`?

Reply:
> Would using serde_json::to_string offer any advantage over format!?

`format!` is not actually JSON. The format is totally undefined.

Reply:
Right, so it seems that `slog` is using it's own `Serialize` trait that's incompatible with serde for whatever reason :man_facepalming:.

> Would using `serde_json::to_string` offer any advantage over `format!`?

Yeah, not pretty but reasonably easy to parse.

Reply:
Ok, so I've updated it to use `serde_json`.